### ISSUE TITLE
[Protocol] 建议 .proto 中的修改 rpc register 为 Register或者其它   

### ISSUE THREAD

**Issue by songzhian** [apache/incubator-skywalking#662]
register  在 c/c++中是关键子， 生成的方法名 编译会报错   没办法友好的支持c/c++程序
建议 .proto 中的修改  register 为 Register或者其它    
 .proto  代码如下
service ApplicationRegisterService {
    rpc register (Application) returns (ApplicationMapping) {
    }
}

**Comment by wu-sheng** [comment]
@OpenSkywalking/pmc and @hanahmily 

This is the a request from c/c++ platform. `register` is definitely a key word in c++. I suggest to change the method name.

@songzhian Is `ApplicationRegisterService#register` the only service which has this problem?

**Comment by wu-sheng** [comment]
## Renaming
* ApplicationRegisterService#register - > ApplicationRegisterService#batchRegister
* Application -> Applications
* Application#applicationCode -> Applications#applicationCodes
* ApplicationMapping -> ApplicationMappings
* ApplicationMapping#application -> ApplicationMappings#applications

New service definition:
```proto
//register service for ApplicationCode, this service is called when service starts.
service ApplicationRegisterService {
    rpc batchRegister (Applications) returns (ApplicationMappings) {
    }
}

message Applications {
    repeated string applicationCodes = 1;
}

message ApplicationMappings {
    repeated KeyWithIntegerValue applications = 1;
}
```

**Comment by peng-yongsheng** [comment]
+1

**Comment by wu-sheng** [comment]
With collector side support, I will do the adjustments as soon as the ASF repo is ready. Sadly, it is still not...

**Comment by wu-sheng** [comment]
This will be fixed by #690 , and released in 5.0.0-alpha.

**Comment by songzhian** [comment]
非常感谢 采纳我提 "c/c++ 的 register关键字问题
但是 应用实例发现服务 proto/DiscoveryService.proto#L11-L12 中的
`service InstanceDiscoveryService {`
`rpc register (ApplicationInstance) returns (ApplicationInstanceMapping) { }`
`}`
还是有 register 关键字
以下是 github的链接
https://github.com/apache/incubator-skywalking/blob/master/apm-network/src/main/proto/DiscoveryService.proto
